lib/WebLedgerClient.js

```diff
@@ -238,34 +248,36 @@ class WebLedgerClient {
    * @returns {Promise<Object>} The result of the send.
    */
   async sendDocument({document, service, headers = {}}) {
-    const requestHeaders = Object.assign(
-      {}, constants.DEFAULT_HEADERS, headers);
     if(!this.ledgerAgent) {
       await this.getAgent();
     }
     const baseURL = this.ledgerAgent.service[service];
-    const serviceApi =
-      create({baseURL, httpsAgent: this.httpsAgent, timeout: this.timeout});
-    const response = await serviceApi.post('/', document, {
-      headers: requestHeaders
-    });
-    if(response.problem) {
+    try {
+      const response = await httpClient.post('', {
+        prefixUrl: baseURL,
+        agent: this.httpsAgent,
+        timeout: this.timeout,
+        json: document,
+        headers: {...DEFAULT_HEADERS, ...headers}
+      });
+      return response.data || null;
+    } catch(e) {
+      const {response} = e;
+      if(!response) {
+        throw e;
+      }
       const error = new WebLedgerClientError(
         'Error sending document.', 'NetworkError',
         {baseURL, service, document});
-      if(response.problem === 'CLIENT_ERROR') {
-        error.details = {
-          baseURL, document, service, error: response.data,
-          status: response.status
-        };
-      } else {
-        error.details = {
-          baseURL, document, service, error: response.originalError
-        };
+      error.details = {
+        baseURL, document, service, error: e
+      };
+      if(response.status >= 400 && response.status < 500) {
+        error.details.status = response.status;
+        error.details.error = e.data || response.data || e;
```

Comment:
Looks to me like this error handling stuff could be DRY'd up, seems like an error handling helper like you worked on elsewhere would be good.

Reply:
issue made: https://github.com/digitalbazaar/web-ledger-client/issues/26